images/entrypoint.sh

```diff
@@ -20,6 +20,13 @@ usage()
     printf "\t--sriov-bin-file=%s\n" $SRIOV_BIN_FILE
 }
 
+exitonsigterm() {
+  echo "sigterm/sigint. exiting"
+  trap - SIGINT SIGTERM # clear trap
+  exit 0
+}
+trap exitonsigterm SIGTERM SIGINT
```

Comment:
I noticed we use `exec` in sriov device plugin to capture sigterm signal, what is the advantage of using a trap function?

Reply:
@zshi-redhat Do you mean calling ```exec sleep 1234``` so that sleep will be pid 1 and receive SIGTERM? I never thought of that! I tested it and it doesn't work as expected.
I think sleep does not handle sigterm but I cannot confirm from [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799.2016edition/utilities/sleep.html) spec.

So, trap allows us to capture a SIGTERM and force exit. If sleep handled a SIGTERM, exec would be fine. I pushed a commit to simplify what I had previously. There was no need to clear the trap.
Without this patch deleting the pod takes 1m30s.
With this patch delete the pod takes 30s.

Reply:
> @zshi-redhat Do you mean calling `exec sleep 1234` so that sleep will be pid 1 and receive SIGTERM? 
Yes
> I never thought of that! I tested it and it doesn't work as expected.
Thanks for testing it!
> I think sleep does not handle sigterm but I cannot confirm from [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799.2016edition/utilities/sleep.html) spec.
> 
> So, trap allows us to capture a SIGTERM and force exit. If sleep handled a SIGTERM, exec would be fine. I pushed a commit to simplify what I had previously. There was no need to clear the trap.
> Without this patch deleting the pod takes 1m30s.
> With this patch delete the pod takes 30s.

Big improvement!
Thanks for the fix!